### PR TITLE
Adding remove plugin.

### PR DESCRIPTION
**Pull request by blockiosaurus**
Adding remove_external_plugin IXes for Assets and Collections. Tested happy paths in rust tests.

### CODE REVIEW

File: programs/mpl-core/src/plugins/utils.rs [hunk 1]
```diff
@@ -415,6 +488,130 @@ pub fn delete_plugin<'a, T: DataBlob>(
...
+                            Ok(authority) => authority,
+                            Err(_) => return Err(MplCoreError::DeserializationError.into()),
+                        }
+                }
+            })
+        {
+            result = Some(i);
+            break;
+        }
+    }
```
Comment (danenbm) [inline programs/mpl-core/src/plugins/utils.rs:557]: Could this block be put in a helper function called `find_external_plugin` that returns `(index, record)`?

If it is not in a helper, it seems like this big block to repeat which makes us re-evaluate whether the record should have the whole key to make it easier, but if the helper just has it once maybe we are more OK with the searching process.

File: programs/mpl-core/src/plugins/utils.rs [hunk 2]
```diff
@@ -415,6 +488,130 @@ pub fn delete_plugin<'a, T: DataBlob>(
...
+
+        header.plugin_registry_offset = new_registry_offset;
+        header.save(account, asset.get_size())?;
+
+        // Move offsets for existing registry records.
+        for record in &mut plugin_registry.external_registry {
+            if plugin_offset < record.offset {
+                record.offset -= serialized_plugin.len()
+            }
+        }
```
Comment (danenbm) [inline programs/mpl-core/src/plugins/utils.rs:607]: I think what is worse here is we need to also move all the data_offset within the external plugin.  This is originally why I wanted to have the `data_offset` at the `ExternalRegistryRecord` level as well, but maybe its not so bad to do it per plugin especially if done in a helper.

**Review by danenbm — CHANGES_REQUESTED** [review]

File: clients/rust/tests/create_with_external_plugins.rs [hunk 3]
```diff
@@ -60,6 +60,7 @@ async fn test_add_lifecycle_hook() {
             name: None,
             uri: None,
             plugins: vec![],
+            external_plugins: vec![],
```
[the remark this answers is not in the document]
Comment (blockiosaurus) [reply clients/rust/tests/create_with_external_plugins.rs:63]: Yeah, originally I was half-assing the tests but probably I should fill these out for real.

File: programs/mpl-core/src/plugins/utils.rs [hunk 4]
```diff
@@ -177,6 +177,79 @@ pub fn fetch_wrapped_plugin<T: DataBlob + SolanaAccount>(
...
+    core: Option<&T>,
+    plugin_key: &ExternalPluginKey,
+) -> Result<(Authority, ExternalPlugin), ProgramError> {
+    let size = match core {
+        Some(core) => core.get_size(),
+        None => {
+            let asset = T::load(account, 0)?;
+
+            if asset.get_size() == account.data_len() {
+                return Err(MplCoreError::PluginNotFound.into());
```
[the remark this answers is not in the document]
Comment (blockiosaurus) [reply programs/mpl-core/src/plugins/utils.rs:192]: Yeah, good call

File: programs/mpl-core/src/plugins/utils.rs (hunk 1 above)
Comment (blockiosaurus) [reply programs/mpl-core/src/plugins/utils.rs:557]: Yes, done!

File: programs/mpl-core/src/plugins/utils.rs (hunk 2 above)
Comment (danenbm) [reply programs/mpl-core/src/plugins/utils.rs:536]: Getting fixed in https://github.com/metaplex-foundation/mpl-core/pull/87

**Review by danenbm — APPROVED** [review]

**Pull request merged**